File: A2OJ/Div2A/Card.cpp
#include<bits/stdc++.h>
#define append push_back
using namespace std;
int main()
{
	string trump,first,second;
	cin >> trump;
	cin >> first >> second;
	map <char,int> p;
	p['6'] = 1;
	p['7'] = 2;
	p['8'] = 3;
	p['9'] = 4;
	p['T'] = 5;
	p['J'] = 6;
	p['Q'] = 7;
	p['K'] = 8;
	p['A'] = 9;
/*
	p.insert(pair<char,int>("6",1));
	p.insert(pair<char,int>("7",2));
	p.insert(pair<char,int>("8",3));
	p.insert(pair<char,int>("9",4));
	p.insert(pair<char,int>("T",5));
	p.insert(pair<char,int>("J",6));
	p.insert(pair<char,int>("Q",7));
	p.insert(pair<char,int>("K",8));
	p.insert(pair<char,int>("A",9));
	*/
	if(first[1].compare(trump) == 0)
	{
		printf("YES\n");
		return 0;
	}
	else if(second[1].compare(trump) == 0)
	{
		printf("NO\n");
	}
	else if(p[first[0]] > p[second[0]])
	{
		printf("YES\n");
		return 0;
	}
	else
	{
		printf("NO\n");
	}
	return 0;
}

File: Codeforces/Problems/Card.py
trump = raw_input()
x = raw_input()
first = x[:2:]
second = x[3::]
d = {
        "6": 1,
        "7": 2,
        "8": 3,
        "9": 4,
        "T": 5,
        "J": 6,
        "Q": 7,
        "K": 8,
        "A": 9
        }
if(first[1] == second[1]):
    if(d[first[0]] > d[second[0]]):
        print "YES"
    else:
        print "NO"
elif(first[1] == trump):
    print "YES"
else:
    print "NO"
